src/ComposerRequireChecker/FileLocator/LocateComposerPackageDirectDependenciesSourceFiles.php

```diff
@@ -12,21 +19,59 @@ public function __invoke(string $composerJsonPath): Generator
 
         $composerJson = json_decode(file_get_contents($composerJsonPath), true);
         $configVendorDir = $composerJson['config']['vendor-dir'] ?? 'vendor';
-        $vendorDirs = array_values(
-            array_map(
-                function (string $vendorName) use ($packageDir, $configVendorDir) {
-                    return $packageDir . '/' . $configVendorDir . '/' . $vendorName;
-                },
-                array_keys($composerJson['require'] ?? [])
-            )
-        );
-
-        foreach ($vendorDirs as $vendorDir) {
+
+        $installedPackages = $this->getInstalledPackages($packageDir . '/' . $configVendorDir);
+
+        $vendorDirs = [];
+        foreach ($composerJson['require'] ?? [] as $vendorName => $vendorRequiredVersion) {
+            $vendorDirs[$vendorName] = $packageDir . '/' . $configVendorDir . '/' . $vendorName;
+        };
+
+        foreach ($vendorDirs as $vendorName => $vendorDir) {
+            $tmp = null;
+
+            # If this package doesn't export its composer.json file then create a temporary one from installed.json
             if (!file_exists($vendorDir . '/composer.json')) {
-                continue;
+                if (array_key_exists($vendorName, $installedPackages)) {
+                    $tmp = $vendorDir . '/composer.json';
+                    file_put_contents($tmp, json_encode($installedPackages[$vendorName]));
```

Comment:
I wasn't keen on creating this file here, but couldn't see a cleaner way without changing `LocateComposerPackageSourceFiles` to accept different parameters, which I thought may be too big of a change. I'd be happy to submit that change as a precursor to this PR though if preferred

Reply:
Do you need to create the file at all? If `installed.json` mirrors the source `composer.json`, then can't you rely on that data structure directly?

Reply:
LocateComposerPackageSourceFiles is responsible for parsing each vendor's composer.json and returning a list of classes.
I'd have to change that class to accept a data structure array instead of a string path. Unless I'm missing something?